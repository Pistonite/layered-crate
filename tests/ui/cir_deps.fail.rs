use layered_crate::layers;

#[layers]
mod src {
    #[depends_on(y)]
    pub extern crate x;
    #[depends_on(x)]
    pub extern crate y;
}

fn main() {
    x::do_something();
}
