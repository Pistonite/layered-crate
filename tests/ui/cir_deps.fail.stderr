error: Circular dependency detected: x -> y -> x
 --> tests/ui/cir_deps.fail.rs:7:5
  |
7 |     #[depends_on(x)]
  |     ^^^^^^^^^^^^^^^^
